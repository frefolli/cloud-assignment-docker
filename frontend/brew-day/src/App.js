
import React from 'react';
import { Route, Routes } from 'react-router';
import NavBar from './components/NavBar';
import Birre from './pages/Birre';
import Home from './pages/Home';
import Impostazioni from './pages/Impostazioni';
import Inventario from './pages/Inventario';
import Ricette from './pages/Ricette';
import Spesa from './pages/Spesa';

const App = () => {
  return (
    <React.Fragment>
      <NavBar />
      <Routes>
        <Route path="/" element={<Home />}/>
        <Route path="Ricette" element={<Ricette />}/>
        <Route path="Birre" element={<Birre />}/>
        <Route path="Inventario" element={<Inventario />}/>
        <Route path="Spesa" element={<Spesa />}/>                
        <Route path="Impostazioni" element={<Impostazioni />}/>
      </Routes>
    </React.Fragment>
  );
}

export default App;