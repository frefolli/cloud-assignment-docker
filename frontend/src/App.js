
import React from 'react';
import { Route, Routes } from 'react-router';
import NavBar from './components/NavBar';
import NavBarThemeManager from './components/NavBarThemeManager';
import Birre from './pages/Birre';
import Home from './pages/Home';
import Impostazioni from './pages/Impostazioni';
import Inventario from './pages/Inventario';
import Ricette from './pages/Ricette';
import Spesa from './pages/Spesa';

export default class App extends React.Component {
  render() {
    return (
      <React.Fragment>
        <NavBarThemeManager>
          <NavBar />
        </NavBarThemeManager>
        <Routes>
          <Route path="/" element={<Home />}/>
          <Route path="/Ricette" element={<Ricette />}/>
          <Route path="/Birre" element={<Birre />}/>
          <Route path="/Inventario" element={<Inventario />}/>
          <Route path="/Spesa" element={<Spesa />}/>                
          <Route path="/Impostazioni" element={<Impostazioni masterCall={() => this.setState({})}/>}/>
        </Routes>
      </React.Fragment>
    );
  };
}