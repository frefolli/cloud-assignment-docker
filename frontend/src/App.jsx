import React, { useState } from 'react';
import { Route, Routes } from 'react-router';
import NavBar from './components/NavBar';
import NavBarThemeManager from './components/NavBarThemeManager';
import Birre from './pages/Birre';
import Home from './pages/Home';
import Impostazioni from './pages/Impostazioni';
import Inventario from './pages/Inventario';
import Ricette from './pages/Ricette';
import Spesa from './pages/Spesa';
import { useSnackbar } from "notistack";

const App = () => {
  const [ flick, setFlick ] = useState(0);
  const { enqueueSnackbar } = useSnackbar();

  return (
      <React.Fragment>
        <NavBarThemeManager>
          <NavBar />
        </NavBarThemeManager>
        <Routes>
          <Route path="/" element={<Home />}/>
          <Route path="/Ricette" element={<Ricette enqueueSnackbar={enqueueSnackbar} />}/>
          <Route path="/Birre" element={<Birre />}/>
          <Route path="/Inventario" element={<Inventario />}/>
          <Route path="/Spesa" element={<Spesa />}/>                
          <Route path="/Impostazioni" element={<Impostazioni masterCall={() => setFlick({})}/>}/>
        </Routes>
      </React.Fragment>
  );
}

export default App;
