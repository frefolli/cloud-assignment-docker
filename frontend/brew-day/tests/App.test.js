import React from 'react';
import { render } from '@testing-library/react';
import App from '../src/App';
import reportWebVitals from '../src/reportWebVitals';
import { act } from "react-test-renderer";
import {BrowserRouter} from "react-router-dom"

global.fetch = jest.fn().mockImplementation(() =>
  Promise.resolve({
    json: () => {
      return Promise.resolve([]);
    },
  })
)

describe('App component', () => {
  test('should render correctly', async () => {
    let theContainer;
    await act(() => {
      const { container } = render(<BrowserRouter><App /></BrowserRouter>);
      theContainer = container;
    })
    expect(theContainer.firstChild).toMatchSnapshot();
  })
});

describe('reportWebVitals', () => {
  test('should launch reportWebVitals', () => {
    reportWebVitals(() => true);
  });
});