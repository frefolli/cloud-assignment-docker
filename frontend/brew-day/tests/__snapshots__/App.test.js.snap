// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly 1`] = `
<div>
  <ul
    id="navBar"
  >
    <li>
      <a
        href="/"
      >
        <img
          id="logo-image"
          src="../icons/logo2.png"
        />
        <span
          id="logo-text"
        >
          Brew Day!
        </span>
      </a>
    </li>
    <li>
      <a
        href="/ricette"
      >
        Ricette
      </a>
    </li>
    <li>
      <a
        href="/birre"
      >
        Birre
      </a>
    </li>
    <li>
      <a
        href="/inventario"
      >
        Inventario
      </a>
    </li>
    <li>
      <a
        href="/spesa"
      >
        Spesa
      </a>
    </li>
  </ul>
</div>
`;
